M365/MDO/MDOThreatPolicyChecker.ps1

```diff
@@ -0,0 +1,840 @@
+            [Parameter(Mandatory = $true)]
+            [string[]]$ExpectedScopes
+        )
+
+        $ValidScope = $true
+        foreach ($ExpectedScope in $ExpectedScopes) {
+            if ($Scopes -contains $ExpectedScope) {
+                Write-Verbose "Scopes $ExpectedScope is present."
+            } else {
+                Write-Host "The following scope is missing: $ExpectedScope" -ForegroundColor Red
+                $ValidScope = $false
+            }
+        }
+        return $ValidScope
+    }
+
+    function Write-HostLog ($message) {
+        if (![string]::IsNullOrEmpty($message)) {
+            $Script:HostLogger = $Script:HostLogger | Write-LoggerInstance $message
+        }
+    }
+
+    SetWriteHostAction ${Function:Write-HostLog}
+
+    $LogFileName = "MDOThreatPolicyChecker"
+    $StartDate = Get-Date
+    $StartDateFormatted = ($StartDate).ToString("yyyyMMddhhmmss")
+    $Script:HostLogger = Get-NewLoggerInstance -LogName "$LogFileName-Results-$StartDateFormatted" -LogDirectory $PSScriptRoot -AppendDateTimeToFileName $false -ErrorAction SilentlyContinue
+
+    $BuildVersion = ""
+
+    Write-Host ("MDOThreatPolicyChecker.ps1 script version $($BuildVersion)") -ForegroundColor Green
+
+    if ($ScriptUpdateOnly) {
+        switch (Test-ScriptVersion -AutoUpdate -VersionsUrl "https://aka.ms/MDOThreatPolicyChecker-VersionsURL" -Confirm:$false) {
+    ($true) { Write-Host ("Script was successfully updated") -ForegroundColor Green }
+    ($false) { Write-Host ("No update of the script performed") -ForegroundColor Yellow }
+            default { Write-Host ("Unable to perform ScriptUpdateOnly operation") -ForegroundColor Red }
+        }
+        return
+    }
+
+    if ((-not($SkipVersionCheck)) -and
+    (Test-ScriptVersion -AutoUpdate -VersionsUrl "https://aka.ms/MDOThreatPolicyChecker-VersionsURL" -Confirm:$false)) {
+        Write-Host ("Script was updated. Please re-run the command") -ForegroundColor Yellow
+        return
+    }
+
+    Write-Host " "
+    Write-Host "Disclaimer:
+
+The sample scripts are not supported under any Microsoft standard support program or service.
+The sample scripts are provided AS IS without warranty of any kind.
+Microsoft further disclaims all implied warranties including, without limitation, any implied warranties of merchantability or of fitness for a particular purpose.
+The entire risk arising out of the use or performance of the sample scripts and documentation remains with you.
+In no event shall Microsoft, its authors, or anyone else involved in the creation, production, or delivery of the scripts be liable for any damages whatsoever
+(including, without limitation, damages for loss of business profits, business interruption, loss of business information, or other pecuniary loss)
+arising out of the use of or inability to use the sample scripts or documentation, even if Microsoft has been advised of the possibility of such damages." -ForegroundColor Yellow
```

Comment:
I'm not sure if we can / should add this disclaimer as part of the script as this is already mentioned as part of the MIT license which you can find here: https://github.com/microsoft/CSS-Exchange?tab=MIT-1-ov-file.

@bill-long @dpaulson45 what's your take on this?

Reply:
This should be removed.